drone_localization/core/models/ConvNext/retrieval_utils.py

```
@@ -1,67 +1,42 @@
-    project_root = os.path.dirname(os.path.dirname(os.path.dirname(current_dir)))
-    dirname = os.path.join(project_root, "weights")
-    # Находим сохраненную модель
+def load_network(name, opt, weights_dir):
+    # weights_dir уже передан, не вычисляем заново
     last_model_name = "net_152.pth"
-
-    # Загружаем конфигурацию
-    config_path = os.path.join(dirname, "opts.yaml")
+    config_path = os.path.join(weights_dir, "opts.yaml")
     with open(config_path, "r") as stream:
         config = yaml.load(stream, Loader=yaml.FullLoader)
```

Ruff (0.15.12)

[error] Probable use of unsafe loader `FullLoader` with `yaml.load`. Allows instantiation of arbitrary objects. Consider `yaml.safe_load`.

(S506)